### PR TITLE
Refactor using native ones

### PR DESCRIPTION
**Pull request by SamuelWei**
Implements #163

Waiting for #172 to add strict types

### CODE REVIEW

File: src/Enum/Role.php [hunk 1]
```diff
@@ -23,8 +23,8 @@
 /**
  * @psalm-immutable
  */
-class Role extends Enum
+enum Role: String
```
Comment (FelixJacobi) [inline src/Enum/Role.php:26]:
```suggestion
enum Role: string
```

From perspective: Less confusing 😄 .

File: src/BigBlueButton.php [hunk 2]
```diff
@@ -114,7 +114,7 @@ class BigBlueButton
      *
      * @throws ConfigException
      */
-    public function __construct(?string $baseUrl = null, ?string $secret = null, ?TransportInterface $transport = null, string $hashingAlgorithm = HashingAlgorithm::SHA_1)
+    public function __construct(?string $baseUrl = null, ?string $secret = null, ?TransportInterface $transport = null, HashingAlgorithm $hashingAlgorithm = HashingAlgorithm::SHA_1)
```
Comment (FelixJacobi) [inline src/BigBlueButton.php:117]: ⚠️ Note: We must note that down as a BC break when releasing 6.0.

**Review by FelixJacobi — APPROVED** [review]
Conflicts, very nice otherwise.

File: src/Parameters/JoinMeetingParameters.php [hunk 3]
```diff
@@ -120,7 +122,7 @@ class JoinMeetingParameters extends UserDataParameters
      */
     protected $excludeFromDashboard;
 
-    public function __construct(string $meetingID, string $fullName, string $role)
+    public function __construct(string $meetingID, string $fullName, Role $role)
```
Comment (SamuelWei) [inline src/Parameters/JoinMeetingParameters.php:125]: ⚠️ Note: We must note that down as a BC break when releasing 6.0.

**Comment by SamuelWei** [comment]
> Conflicts, very nice otherwise.

fixed

**Review by FelixJacobi — APPROVED** [review]

**Pull request merged**